service/invoker/invoker.go

```diff
@@ -159,32 +157,31 @@ func (i *Invoker) Script(height uint64, script []byte, arguments []cadence.Value
 	}
 
 	// Initialize the virtual machine context with the given block header so
-	// that parameters related to the block are available from within the script.
+	// that parameters related to the block are available from within the
+	// script.
 	ctx := fvm.NewContext(fvm.WithBlockHeader(header))
 
-	// Initialize the read function. We use a shared cache between all heights
-	// here. It's a smart cache, which means that items that are accessed often
-	// are more likely to be kept, regardless of height. This allows us to put
-	// an upper bound on total cache size while using it for all heights.
-	read := readRegister(i.index, i.cache, height)
-
-	// Initialize the view of the execution state on top of the ledger by
-	// using the read function at a specific commit.
-	view := delta.NewView(read)
+	// Initialize the storage snapshot. We use a shared cache between all
+	// heights here. It's a smart cache, which means that items that are
+	// accessed often are more likely to be kept, regardless of height. This
+	// allows us to put an upper bound on total cache size while using it for
+	// all heights.
+	storageSnapshot := snapshot.NewReadFuncStorageSnapshot(
+		readRegister(i.index, i.cache, height))
 
 	// Initialize the procedure using the script bytes and the encoded
 	// Cadence parameters.
 	proc := fvm.Script(script).WithArguments(args...)
 
 	// The script procedure is then run using the Flow virtual machine and all
 	// the constructed contextual parameters.
-	err = i.vm.Run(ctx, proc, view)
+	_, output, err := i.vm.Run(ctx, proc, storageSnapshot)
```

Comment:
Do we need a TODO to use `Run2`?

Reply:
i've already renamed RunV2 to Run in the flow-go code base

Reply:
IIRC we renamed RunV2 back to Run at some point